### PR TITLE
Apply Apache License header for Tcl files

### LINKED ISSUE #547: [NEW] Apply Apache License header for Tcl files
This is a follow up to #542.

Finally I understand how to apply Apache License header for Tcl files.

### PR DESCRIPTION
**Pull request by tisonkun**
This closes #547.

Signed-off-by: tison <wander4096@gmail.com>

### CODE REVIEW

**Comment by tisonkun** [comment]
cc @Hexiaoqiao @git-hulk

**Review by git-hulk — DISMISSED** [review]
LGTM

**Comment by git-hulk** [comment]
cc @ShooterIT

**Comment by tisonkun** [comment]
Hold as most tcl files are copied and modified from Redis files. We should take care of BSD-3 Clause License requirements.

**Review by Hexiaoqiao — COMMENTED** [review]
The new header about License and Copyright annotation looks good to me.
I am not sure if this is complete, any source file is omitted?
NOTE: This round review is random checking, not checking every source file one by one.

**Comment by tisonkun** [comment]
@Hexiaoqiao in the last turn #542 I've checked all other files except these tcl files. And this turn I dealt with them.

**Comment by Hexiaoqiao** [comment]
Thanks. +1.

**Review by Hexiaoqiao — APPROVED** [review]

**Review by git-hulk — APPROVED** [review]
+1

File: licenses/LICENSE.redis [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Copyright (c) 2006-2020, Salvatore Sanfilippo
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
```
Comment (vongosling) [inline licenses/LICENSE.redis:6]: We should care about the redistribution requirements here. Do we check the compatibility with ASL v2?

Comment (tisonkun) [reply licenses/LICENSE.redis:6]: > ASL v2

Do you mean Apache License 2.0?

> compatibility

This is a typical 3-Clause BSD License which is compatible with Apache License 2.0.

* https://www.apache.org/legal/resolved.html#category-a

**Pull request merged**